File: app-mvvm/src/main/java/com/xuie/imaginary/gui/nets/video/VideoMainFragment.java
package com.xuie.imaginary.gui.nets.video;


import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.xuie.imaginary.R;
import com.xuie.imaginary.base.BaseFragment;
import com.xuie.imaginary.databinding.FragmentVideoMainBinding;

import java.util.Arrays;
import java.util.List;

/**
 * A simple {@link Fragment} subclass.
 *
 * @author xuie
 */
public class VideoMainFragment extends BaseFragment {

    public static VideoMainFragment getInstance() {
        return new VideoMainFragment();
    }

    private FragmentVideoMainBinding mBinding;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_video_main;
    }

    @Override
    protected void onInit(@Nullable Bundle savedInstanceState) {
        mBinding = getDataBinding();

        mBinding.viewPager.setAdapter(new MyPagerAdapter(getChildFragmentManager()));
        mBinding.tabs.setupWithViewPager(mBinding.viewPager);
    }

    @Override
    protected void lazyInitData() {

    }

    private class MyPagerAdapter extends FragmentPagerAdapter {
        List<String> channelName = Arrays.asList(getResources().getStringArray(R.array.video_channel_name));
        List<String> channelId = Arrays.asList(getResources().getStringArray(R.array.video_channel_id));

        MyPagerAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            return VideosFragment.getInstance(channelName.get(position), channelId.get(position));
        }

        @Override
        public int getCount() {
            return channelId.size();
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return channelName.get(position);
        }
    }
}

File: app-mvvm-kotlin/src/main/kotlin/xuk/imaginary/gui/nets/video/VideoMainFragment.kt
package xuk.imaginary.gui.nets.video

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import xuk.imaginary.R
import xuk.imaginary.databinding.FragmentVideoMainBinding
import java.util.*

/**
 * A simple [Fragment] subclass.
 *
 * @author Jie Xu
 */
class VideoMainFragment : Fragment() {

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
    val binding = FragmentVideoMainBinding.inflate(inflater, container, false)
    binding.viewPager.adapter = MyPagerAdapter(childFragmentManager)
    binding.tabs.setupWithViewPager(binding.viewPager)
    return binding.root
  }

  private inner class MyPagerAdapter internal constructor(fm: FragmentManager) : FragmentPagerAdapter(fm) {
    internal var channelName = Arrays.asList(*resources.getStringArray(R.array.video_channel_name))
    internal var channelId = Arrays.asList(*resources.getStringArray(R.array.video_channel_id))

    override fun getItem(position: Int): Fragment {
      return VideosFragment.getInstance(channelName[position], channelId[position])
    }

    override fun getCount(): Int {
      return channelId.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
      return channelName[position]
    }
  }

  companion object {

    val instance: VideoMainFragment
      get() = VideoMainFragment()
  }
}
